core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            auto strVal = value.as<std::string>();
+            type = "int";
```

Comment:
I would prefer to not add support for `int` uniforms since they appear to
not be supported in WebGL, and I'd like to keep parity so we don't have
files that work on one platform but not the other (that's clearly pretty
aspirational right now, but we can at least avoid making the gap wider :)

On Wed, Sep 23, 2015 at 3:36 PM, Hannes Janetzek <notifications@github.com>
wrote:

> In core/src/scene/sceneLoader.cpp
> <https://github.com/tangrams/tangram-es/pull/294#discussion_r40247294>:
>
> > @@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
> >      return out;
> >  }
> >
> > +StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
> > +    std::string type = "";
> > +    std::vector<UniformValue> uniformValues;
> > +    if (value.IsScalar()) { //float, int (bool), string (texture)
> > +        UniformValue uVal;
> > +        try {
> > +            uVal = value.as<float>();
> > +            type = "float";
> > +        } catch (const BadConversion& e) {
> > +            auto strVal = value.as<std::string>();
> > +            type = "int";
>
> cpp-yaml converts quite a lot of things to boolean: {"y", "n"}, {"yes",
> "no"}, {"true", "false"}, {"on", "off"} http://yaml.org/type/bool.html.
> Though chances are low that someone names textures like this :)
>
> @bcamper <https://github.com/bcamper> don't need to doesn't mean we may
> not add support for int uniforms while we are at it. Or is it about strict
> feature parity?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/294/files#r40247294>.
>